### PR TITLE
upgrade react-Iintl to v5

### PR DESCRIPTION
**Pull request by wolverineks**
closes `app settings screen is not refreshed after language change`
enables migrating from `injectIntl(Component)` -> `const intl = useIntl()`

tested on iPhone 12 simulator, Pixel 3 XL emulator

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -66,7 +66,7 @@
     "node-fetch": "^2.6.1",
     "p-limit": "3.1.0",
     "react": "16.13.1",
-    "react-intl": "^2.8.0",
+    "react-intl": "^5.17.4",
```
Comment (v-almonacid) [inline package.json]: forgot to mention: we are now using exact versions for all our dependencies. Could you please set it to `"react-intl": "5.17.4"`?

**Comment by wolverineks** [comment]
so, it looks like there are some polyfills that react-intl recommends, but everything seems to work even without them
the only thing i did see were some errors in the logs about missing locale data, but all the translations looked like they worked

additionally, the polyfills dont have anything czech

i dunno, what do you think? include them or ignore it for now?


```
// @flow
import '@formatjs/intl-locale/polyfill'

import '@formatjs/intl-numberformat/polyfill'
import '@formatjs/intl-numberformat/locale-data/de.js'
import '@formatjs/intl-numberformat/locale-data/en.js'
import '@formatjs/intl-numberformat/locale-data/es.js'
import '@formatjs/intl-numberformat/locale-data/fr.js'
import '@formatjs/intl-numberformat/locale-data/id.js'
import '@formatjs/intl-numberformat/locale-data/it.js'
import '@formatjs/intl-numberformat/locale-data/ja.js'
import '@formatjs/intl-numberformat/locale-data/ko.js'
import '@formatjs/intl-numberformat/locale-data/nl.js'
import '@formatjs/intl-numberformat/locale-data/pt.js'
import '@formatjs/intl-numberformat/locale-data/ru.js'
import '@formatjs/intl-numberformat/locale-data/zh.js'

import '@formatjs/intl-datetimeformat/polyfill'
import '@formatjs/intl-datetimeformat/locale-data/de.js'
import '@formatjs/intl-datetimeformat/locale-data/en.js'
import '@formatjs/intl-datetimeformat/locale-data/es.js'
import '@formatjs/intl-datetimeformat/locale-data/fr.js'
import '@formatjs/intl-datetimeformat/locale-data/id.js'
import '@formatjs/intl-datetimeformat/locale-data/it.js'
import '@formatjs/intl-datetimeformat/locale-data/ja.js'
import '@formatjs/intl-datetimeformat/locale-data/ko.js'
import '@formatjs/intl-datetimeformat/locale-data/nl.js'
import '@formatjs/intl-datetimeformat/locale-data/pt.js'
import '@formatjs/intl-datetimeformat/locale-data/ru.js'
import '@formatjs/intl-datetimeformat/locale-data/zh.js'
```

**Comment by wolverineks** [comment]
closes #1313

**Comment by vantuz-subhuman** [comment]
[ch7305]

**Comment by v-almonacid** [comment]
> so, it looks like there are some polyfills that react-intl recommends, but everything seems to work even without them
> the only thing i did see were some errors in the logs about missing locale data, but all the translations looked like they worked
> 
> additionally, the polyfills dont have anything czech
> 
> i dunno, what do you think? include them or ignore it for now?
> 
> ```
> // @flow
> import '@formatjs/intl-locale/polyfill'
> 
> import '@formatjs/intl-numberformat/polyfill'
> import '@formatjs/intl-numberformat/locale-data/de.js'
> import '@formatjs/intl-numberformat/locale-data/en.js'
> import '@formatjs/intl-numberformat/locale-data/es.js'
> import '@formatjs/intl-numberformat/locale-data/fr.js'
> import '@formatjs/intl-numberformat/locale-data/id.js'
> import '@formatjs/intl-numberformat/locale-data/it.js'
> import '@formatjs/intl-numberformat/locale-data/ja.js'
> import '@formatjs/intl-numberformat/locale-data/ko.js'
> import '@formatjs/intl-numberformat/locale-data/nl.js'
> import '@formatjs/intl-numberformat/locale-data/pt.js'
> import '@formatjs/intl-numberformat/locale-data/ru.js'
> import '@formatjs/intl-numberformat/locale-data/zh.js'
> 
> import '@formatjs/intl-datetimeformat/polyfill'
> import '@formatjs/intl-datetimeformat/locale-data/de.js'
> import '@formatjs/intl-datetimeformat/locale-data/en.js'
> import '@formatjs/intl-datetimeformat/locale-data/es.js'
> import '@formatjs/intl-datetimeformat/locale-data/fr.js'
> import '@formatjs/intl-datetimeformat/locale-data/id.js'
> import '@formatjs/intl-datetimeformat/locale-data/it.js'
> import '@formatjs/intl-datetimeformat/locale-data/ja.js'
> import '@formatjs/intl-datetimeformat/locale-data/ko.js'
> import '@formatjs/intl-datetimeformat/locale-data/nl.js'
> import '@formatjs/intl-datetimeformat/locale-data/pt.js'
> import '@formatjs/intl-datetimeformat/locale-data/ru.js'
> import '@formatjs/intl-datetimeformat/locale-data/zh.js'
> ```

I wouldn't include them if they are not strictly necessary. Is there any section in react-intl docs that refers to this?
I just saw this https://formatjs.io/docs/react-intl#react-native

**Comment by v-almonacid** [comment]
Also, some strings are not being correctly parsed and translated. Might be a change in the recent API?

image

File: package.json [hunk 2]
```diff
@@ -32,7 +32,10 @@
     "@cardano-foundation/ledgerjs-hw-app-cardano": "2.2.1",
     "@emurgo/cip14-js": "2.0.0",
     "@emurgo/cip4-js": "1.0.5",
-    "@emurgo/react-native-haskell-shelley": "1.1.4",
+    "@emurgo/react-native-haskell-shelley": "1.1.3",
```
Comment (v-almonacid) [inline package.json]: We should probably need to rebase since this is being downgraded

**Pull request closed without merging**